File: java/hamming/src/main/java/Hamming.java
public class Hamming {

    private String left;
    private String right;

    public Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("strands must be of equal length");
        }
        this.left = leftStrand;
        this.right = rightStrand;
    }

    public int getHammingDistance() {
        int distance = 0;
        for(int i = 0; i < this.left.length(); i++) {
            if (left.charAt(i) != right.charAt(i)) {
                distance++;
            }
        }
        return distance;
    }
}

File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
  if #a ~= #b then
    -- return false
    error("strands must be of equal length")
  end
  
  local distance = 0
  for i = 1, #a do
    local al = a:sub(i, i)
    local bl = b:sub(i, i)
    if al ~= bl then
      distance = distance + 1
    end
  end
  return distance
end

return Hamming
